### PR TITLE
Button Logics repaired / improved

### PR DESCRIPTION
**Pull request by amyria3**

### CODE REVIEW

**Comment by NoHara42** [comment]
@amyria3 what issues does this relate to?

File: src/components/Button.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ interface ButtonProps {
   height: "default" | "small"
   label: string;
   onClick: (event: React.FormEvent<any>) => void;
-  disabled: boolean; //state
+  disabled?: true; //state
```
Comment (NoHara42) [inline src/components/Button.tsx:11]: are you sure this is supposed to be `true`? What's the use of hard coding values of a prop here?

**Review by NoHara42 — CHANGES_REQUESTED** [review]
take a look at my comment

**Comment by amyria3** [comment]
Maybe there is no additional value... I just was so so happy that I understood how dan writes it 🥇 
and maybe its too late to change it now, to many people would be affected

File: src/components/Button.tsx (hunk 1 above)
Comment (dnmct) [reply src/components/Button.tsx:11]: this defo needs to be boolean and not true, the logic may work but its kind of a bad practice

better solution:
```
disabled?: boolean

function Button({
  type, 
  height, 
  label, 
  disabled = false
}:
ButtonProps){
```

this makes the prop optional to pass in and when you not pass it it's not disabled

**Review by dnmct — CHANGES_REQUESTED** [review]

**Comment by amyria3** [comment]
Yes, I made a patch for it once I understood the syntax you use and its beauty :) 
https://github.com/devhausleipzig/camp-8-midterm-group-2/blob/amyria3-patch-1/src/components/Button.tsx

**Pull request merged**